posthog/api/element.py

```diff
@@ -0,0 +1,61 @@
+from rest_framework import request, response, serializers, viewsets
+from rest_framework.decorators import action
+from posthog.models import Element, Team
+from django.db.models import QuerySet
+
+class ElementSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Element
+        fields = ['text', 'tag_name', 'attr_class', 'href', 'attr_id', 'nth_child', 'nth_of_type', 'attributes', 'order']
+
+
+class ElementViewSet(viewsets.ModelViewSet):
+    queryset = Element.objects.all()
+    serializer_class = ElementSerializer
+
+    def get_queryset(self) -> QuerySet:
+        queryset = super().get_queryset()
+
+        return queryset.filter(group__team=self.request.user.team_set.get())
+ 
+    @action(methods=['GET'], detail=False)
+    def values(self, request: request.Request) -> response.Response:
+        key = request.GET.get('key')
+        params = []
+        where = ''
+
+        # Make sure key exists, otherwise could lead to sql injection lower down
+        if key not in self.serializer_class.Meta.fields:
+            return response.Response([])
+
+        if request.GET.get('value'):
+            where = ' AND "posthog_element"."{}" LIKE %s'.format(key)
+            params.append('%{}%'.format(request.GET['value']))
+
+        # This samples a bunch of elements with that property, and then orders them by most popular in that sample
+        # This is much quicker than trying to do this over the entire table
+        values = Element.objects.raw("""
```

Comment:
There maybe be some reusability here with /path/elements endpoint in paths. Made ticket for followup here #954 954